### PR TITLE
feat: handle not found profile

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/rnd-dev-team#1786

Handle not found found public profile.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### How to test

1. [Invalid regen address] https://deploy-preview-2136--regen-marketplace.netlify.app/profiles/regen1c8ythqpcn5pf4qfg99rwx0djnf0fh4xgztg/portfolio

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by flagrede** [comment]
@erikalogie you can check the not found profile case here: https://deploy-preview-2136--regen-marketplace.netlify.app/profiles/regen1c8ythqpcn5pf4qfg99rwx0djnf0fh4xgztg/portfolio

File: web-marketplace/src/pages/EcocreditsByAccount/EcocreditsByAccount.tsx [hunk 1]
```diff
@@ -33,6 +34,7 @@ export const EcocreditsByAccount = (): JSX.Element => {
 
   const { address, party } = useProfileData();
   const { avatarImage, backgroundImage } = getUserImages({ party });
+  const isProfileNotFound = !address && !party;
```
Comment (blushi) [inline web-marketplace/src/pages/EcocreditsByAccount/EcocreditsByAccount.tsx:37]: when the page's loading, we can see the "profile not found" page for a few ms
we should probably add a loader

**Comment by erikalogie** [comment]
looks good!

**Review by blushi — COMMENTED** [review]
Somehow I'm now seeing the "profile not found" for a few ms first, then the loader and finally "profile not found"

**Comment by flagrede** [comment]
> Somehow I'm now seeing the "profile not found" for a few ms first, then the loader, and finally "profile not found"

@blushi It was because of the CSRF token loading I added it in the fetching loading logic, should be better now.
I did not see this locally because the request might have been too fast.

**Review by blushi — APPROVED** [review]

**Pull request merged**